Incompatibility with optparse-applicative-0.18.1.0 + ansi-wl-pprint-0.6.9

If we compile toysolver 0.8.1 with `optparse-applicative-0.18.1.0` + `ansi-wl-pprint-0.6.9`, it fails with the following error:

```
/Users/sakai/src/toysolver/app/toyconvert.hs:198:22: error:
    • Couldn't match type ‘PP.Doc’
                     with ‘prettyprinter-1.7.1:Prettyprinter.Internal.Doc
                             prettyprinter-ansi-terminal-1.1.3:Prettyprinter.Render.Terminal.Internal.AnsiStyle’
      Expected: Options.Applicative.Help.Pretty.Doc
        Actual: PP.Doc
      NB: ‘prettyprinter-1.7.1:Prettyprinter.Internal.Doc’
            is defined in ‘Prettyprinter.Internal’
                in package ‘prettyprinter-1.7.1’
          ‘PP.Doc’
            is defined in ‘Text.PrettyPrint.ANSI.Leijen.Internal’
                in package ‘ansi-wl-pprint-0.6.9’
    • In the first argument of ‘Just’, namely ‘supportedFormatsDoc’
      In the first argument of ‘footerDoc’, namely
        ‘(Just supportedFormatsDoc)’
      In the second argument of ‘(<>)’, namely
        ‘footerDoc (Just supportedFormatsDoc)’
    |
198 |   <> footerDoc (Just supportedFormatsDoc)
    |                      ^^^^^^^^^^^^^^^^^^^
```

| |ansi-wl-pprint-0.6.9|ansi-wl-pprint-1.0.2|
|-|--------------------|--------------------|
|optparse-applicative-0.17.1.0| OK | OK |
|optparse-applicative-0.18.1.0| NG (above error) | OK |

--- a/toysolver.cabal
+++ b/toysolver.cabal
@@ -123,6 +123,11 @@ Flag ExtraBoundsChecking
   Manual: True
   Default: False
 
+Flag optparse-applicative-018
+  Description: use optparse-applicative >=0.18
+  Manual: False
+  Default: False
+
 source-repository head
   type:     git
   location: git://github.com/msakai/toysolver.git
@@ -553,17 +558,23 @@ Executable toyconvert
   HS-Source-Dirs: app
   Build-Depends:
     base,
-    ansi-wl-pprint,
     bytestring,
     bytestring-builder,
     data-default-class,
     filepath,
     MIP,
-    optparse-applicative,
     pseudo-boolean,
     scientific,
     text,
     toysolver
+  if flag(optparse-applicative-018)
+    Build-Depends:
+      optparse-applicative >=0.18,
+      ansi-wl-pprint >=1.0
+  else
+    Build-Depends:
+      optparse-applicative <0.18,
+      ansi-wl-pprint
   Default-Language: Haskell2010
   Other-Extensions: CPP
   GHC-Options: -rtsopts

